### PR TITLE
Resolved rotation issue in TwoHandManipulatable script

### PR DESCRIPTION
**Pull request by shubhamarora28**
Overview
---
Issue:
If we change the rotation axis constraint at run time, it does not make any difference in the behaviour.

Fix: Reinitialize the rotate logic with the new rotation axis constraint every time the rotation constraint property is set.

Sample:
I have duplicated the TwoHandManipulationTest scene as TwoHandManipulationTest_Fix scene and have added the following game object into the scene content:
- Model_Bucky_Rotation_Constraints_Test: This has a test script(TestRotationConstraints) attached to it and is intended to depict the solution to the issue.

Changes
---
Changed TwoHandManipultable.cs
Added TestRotationConstraints.cs
Added TwoHandManipulationTest_Fix.unity

### CODE REVIEW

File: Assets/HoloToolkit-Examples/Input/Scripts/TestRotationConstraints.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        /// </summary>
+        private TwoHandManipulatable twoHandManipulatable;
+
+        private const string DescriptionPrefix = "Rotate\n";
+        private const string XAxisConstraint = "X axis Constraint";
+        private const string YAxisConstraint = "Y axis Constraint";
+        private const string DescriptionPostfix = "\nTap on the model to toggle between X and Y constraints";
+
+        // Use this for initialization
+        void Start()
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/Input/Scripts/TestRotationConstraints.cs]: nit missing private access modifier.

File: Assets/HoloToolkit-Examples/Input/Scripts/TestRotationConstraints.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    /// attached to make the rotation axis constraint toggle between X and Y axis on tapping the object.
+    /// </summary>
+    [RequireComponent(typeof(TwoHandManipulatable))]
+    public class TestRotationConstraints : MonoBehaviour, IInputClickHandler
+    {
+        /// <summary>
+        /// The TextMesh game object showing the description of the manipulation.
+        /// </summary>
+        [SerializeField]
+        private TextMesh descriptionText;
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/Input/Scripts/TestRotationConstraints.cs]: initialize with null

**Comment by davidkline-ms** [comment]
> You can either rename it as an example and remove the fix suffix, or remove the test scene all together. Up to you.

I'm good with not adding a new test scene.

File: Assets/HoloToolkit-Examples/Input/Scripts/TestRotationConstraints.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/HoloToolkit-Examples/Input/Scripts/TestRotationConstraints.cs]: If not adding a fix specific test scene, I presume this file can be left out of the PR.

**Comment by shubhamarora28** [comment]
@davidkline-ms The test scene I had created has a game object(Model_Bucky_Rotation_Constraints_Test) that depicts the capability of TwoHandManipulatable to modify the rotation constraint at run time.
Instead of adding this as a new scene just for this fix, I was thinking if I can replace the original TwoHandManipulationTest scene with the one that I have added. Will that be okay?

**Comment by SimonDarksideJ** [comment]
Does this fix also need applying to vNext ? @davidkline-ms @StephenHodgson

**Comment by StephenHodgson** [comment]
We haven't ported it yet.

**Comment by davidkline-ms** [comment]
@shubhamarora28, as long as we don't lose functionality in the existing example scene, I think that sounds fine.

Thanks!

**Comment by davidkline-ms** [comment]
> Does this fix also need applying to vNext ? @davidkline-ms @StephenHodgson

> We haven't ported it yet.

It should definitely be added to #2465 for tracking.

**Comment by shubhamarora28** [comment]
@StephenHodgson @davidkline-ms Pushed a commit with all the changes.

**Pull request merged**